extensions/src/org/pentaho/platform/web/http/api/resources/services/FileService.java

```diff
@@ -743,7 +745,12 @@ public void doCopyFiles( String pathId, Integer mode, String params ) {
     CopyFilesOperation copyFilesOperation =
       new CopyFilesOperation( getRepository(), getRepoWs(), Arrays.asList( sourceFileIds ), path, mode );
 
-    copyFilesOperation.execute();
+    try {
+      copyFilesOperation.execute();
+    } catch ( UnifiedRepositoryAccessDeniedException e ) {
+      // let's convert it to an appropriate one for REST resource
+      throw new IllegalArgumentException();
```

Comment:
You should pass on the original exception here - this is swallowing what the actual error was.